packages/api-explorer/src/components/ExploreType/exploreUtils.ts

```diff
@@ -0,0 +1,137 @@
+ SOFTWARE.
+
+ */
+
+import { IconNames } from '@looker/components'
+import { IType, TypeOfType, typeOfType } from '@looker/sdk-codegen'
+
+/**
+ * Get the type or element type if this type is a collection
+ * @param value type to pick
+ */
+export const pickType = (value: IType) => {
+  const typed = typeOfType(value)
+  switch (typed) {
+    case TypeOfType.Intrinsic:
+    case TypeOfType.Complex:
+      return value
+    default:
+      return value.elementType!
+  }
+}
+
+/**
+ * Is this an expandable node?
+ * @param level is the current level of the hierarchy
+ * @param maxDepth is the maximum depth to expanded nested types. -1 = all (default), 0 = no expansion
+ */
+export const expandable = (level: number, maxDepth = -1) =>
+  maxDepth === -1 || level <= maxDepth
+
+/**
+ * Gets the properties for the "exploring" type
+ * @param value type to pick
+ */
+export const pickTypeProps = (value: IType) => pickType(value).properties
+
+/**
+ * Get the link prefix for the type
+ * @param value type to link to
+ */
+export const typeLinkPrefix = (value: IType) => {
+  const typed = typeOfType(value)
+  switch (typed) {
+    case TypeOfType.Hash:
+      return 'Hash['
+    default:
+      return ''
+  }
+}
+
+/**
+ * Get the link suffix for the type
+ * @param value type to link to
+ */
+export const typeLinkSuffix = (value: IType) => {
+  const typed = typeOfType(value)
+  switch (typed) {
+    case TypeOfType.Array:
+    case TypeOfType.DelimArray:
+      return '[]'
+    case TypeOfType.Hash:
+      return ']'
+    default:
+      return ''
+  }
+}
+
+interface TypedIcon {
+  icon: IconNames
+  title: string
+}
+
+/**
+ * Determines the icon name for the type
+ * @param value to iconize
+ */
+export const typeIcon = (value: IType): TypedIcon => {
+  switch (value.className) {
+    case 'ArrayType':
+      return { icon: 'ChartSingleRecord', title: value.jsonName }
+    case 'DelimArrayType':
+      return { icon: 'ChartSingleRecord', title: value.jsonName }
+    case 'HashType':
+      return { icon: 'IdeFileManifest', title: value.jsonName }
+    case 'EnumType':
+      return { icon: 'IdeParameter', title: value.jsonName }
+  }
+
+  const type = pickType(value)
+  switch (type.jsonName) {
+    case 'boolean':
+      return { icon: 'FieldYesNo', title: type.jsonName }
+    case 'int64':
+    case 'integer':
+    case 'float':
+    case 'double':
+      return { icon: 'FieldNumber', title: type.jsonName }
+    case 'string':
+    case 'hostname':
+    case 'uuid':
+    case 'ipv4':
+    case 'ipv6':
+      return { icon: 'FieldString', title: type.jsonName }
+    case 'email':
+      return { icon: 'SendEmail', title: type.jsonName }
+    case 'password':
+      return { icon: 'Key', title: type.jsonName }
+    case 'uri':
+      return { icon: 'Link', title: type.jsonName }
+    case 'datetime':
+      return { icon: 'FieldDate', title: type.jsonName }
+    default:
+      return { icon: 'Code', title: type.jsonName }
+  }
+}
```

Comment:
I'm abusing several of these icons because we don't have more appropriate ones yet. Better icons is a TODO item, but they aren't required to land the new type explorer.